code/obj/chem_fire.dm

```diff
@@ -0,0 +1,93 @@
+// fires created by chemical or magical sources
+/obj/chem_fire
+	mouse_opacity = 0
+	anchored = ANCHORED_ALWAYS
+	flags = UNCRUSHABLE
+	layer = NOLIGHT_EFFECTS_LAYER_BASE
+	plane = PLANE_ABOVE_LIGHTING
+
+	icon = 'icons/effects/fire.dmi'
+	icon_state = "chem_fire-red"
+
+	blend_mode = BLEND_ADD
+
+	// light source created by the fire
+	var/datum/light/light
+	/// volume of fire to expose to crossing atoms
+	var/volume
+	/// temperature of fire
+	var/temperature
+
+/obj/chem_fire/New(turf/simulated/T, volume = 125, temperature = FIRE_MINIMUM_TEMPERATURE_TO_EXIST, duration = 3 SECONDS, color = CHEM_FIRE_RED)
+	..()
+	START_TRACKING
+	if (!istype(T) || (locate(/obj/fire_foam) in T))
+		qdel(src)
+		return
+	if (!T.air || (T.air.oxygen < 0.5 MOLES && T.air.toxins < 0.5 MOLES))
+		qdel(src)
+		return
+	if (temperature < FIRE_MINIMUM_TEMPERATURE_TO_EXIST || volume <= 1)
+		qdel(src)
+		return
+	var/area/A = get_area(T)
+	if (A.sanctuary)
+		return
+	for (var/obj/chem_fire/existing_fire in T)
+		if (existing_fire == src)
+			continue
+		src.temperature = max(src.temperature, existing_fire.temperature)
+		qdel(existing_fire)
+		break
+
+	SPAWN(duration)
+		qdel(src)
```

Comment:
since you're running processing on these, should track remaining duration and kill in the process instead of using a SPAWN here